src/game/Chat/Level2.cpp

```diff
@@ -2317,6 +2317,165 @@ bool ChatHandler::HandleNpcShowLootCommand(char* /*args*/)
     return true;
 }
 
+// show detailed information of creature formation if exist
+bool ChatHandler::HandleNpcGroupInfoCommand(char* /*args*/)
+{
+    Creature* creature = getSelectedCreature();
+
+    if (!creature)
+    {
+        SendSysMessage(LANG_SELECT_CREATURE);
+        SetSentErrorMessage(true);
+        return false;
+    }
+
+    auto gData = creature->GetCreatureGroup();
+    if (!gData)
+    {
+        SendSysMessage("Creature is not in group");
+        return true;
+    }
+
+    PSendSysMessage("Group id[%u]", gData->GetGroupId());
+    PSendSysMessage("Group name: %s", gData->GetGroupEntry().Name.c_str());
+
+    Unit* master = nullptr;
+    if (gData->GetFormationData())
+        master = gData->GetFormationData()->GetMaster();
+
+    if (master)
+    {
+        if (master == creature)
+            SendSysMessage("Creature is formation leader");
+        else if (creature->GetFormationSlot())
+            PSendSysMessage("Creature is in formation slot");
+        else
+            SendSysMessage("Error: unable to retrieve the slot of the creature.");
+    }
+
+    SendSysMessage(gData->to_string().c_str());
```

Comment:
Non standard way of things. Compiler will scream. Do PSendSys and format specifier %s and pass it as arg 1.